### PR TITLE
Add support for destroying signing key versions

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes GH-349

Part of https://github.com/google/exposure-notifications-server/issues/913. 

**Release Note**

```release-note
Add support for destroying signing key versions
```

/assign @mikehelmick

### CODE REVIEW

File: pkg/controller/realmkeys/render.go [hunk 1]
```diff
@@ -34,12 +34,12 @@ func (c *Controller) renderShow(ctx context.Context, w http.ResponseWriter, r *h
 
 	m["supportsPerRealmSigning"] = c.db.SupportsPerRealmSigning()
 	if c.db.SupportsPerRealmSigning() {
-		c.logger.Infof("listing signing keys")
 		keys, err := realm.ListSigningKeys(c.db)
-		c.logger.Infow("list result", "error", err, "keys", keys)
 		if err != nil {
 			controller.InternalError(w, r, c.h, err)
+			return
```
Comment (sethvargo) [inline pkg/controller/realmkeys/render.go:40]: Twas bug

File: pkg/database/realm.go [hunk 2]
```diff
@@ -513,3 +520,51 @@ func (r *Realm) CreateSigningKeyVersion(ctx context.Context, db *Database) (stri
...
+
+		// Delete the signing key from the key manager - we want to do this in the
+		// transaction so, if it fails, we can rollback and try again.
+		if err := manager.DestroyKeyVersion(ctx, signingKey.KeyID); err != nil {
+			return fmt.Errorf("failed to destroy signing key in key manager: %w", err)
+		}
+
+		// Successfully deleted from the key manager, now remove the record.
+		if err := tx.Delete(&signingKey).Error; err != nil {
+			return fmt.Errorf("failed to delete signing key from database: %w", err)
```
Comment (mikehelmick) [inline pkg/database/realm.go:561]: but... key material was destroyed. I would be sure to add that to the error message.

File: pkg/database/realm.go [hunk 3]
```diff
@@ -513,3 +520,51 @@ func (r *Realm) CreateSigningKeyVersion(ctx context.Context, db *Database) (stri
...
+			return fmt.Errorf("failed to load signing key: %w", err)
+		}
+
+		if signingKey.Active {
+			return fmt.Errorf("cannot destroy active signing key")
+		}
+
+		// Delete the signing key from the key manager - we want to do this in the
+		// transaction so, if it fails, we can rollback and try again.
+		if err := manager.DestroyKeyVersion(ctx, signingKey.KeyID); err != nil {
```
Comment (mikehelmick) [inline pkg/database/realm.go:555]: unlikely ... but

File: pkg/database/realm.go (hunk 2 above)
Comment (sethvargo) [reply pkg/database/realm.go]: Done

**Pull request merged**